lib/livebook/hubs.ex

```diff
@@ -186,18 +186,67 @@ defmodule Livebook.Hubs do
   ## Example
 
       iex> get_connected_hubs()
-      [%{pid: #PID<0.178.0>, hub: %Enterprise{}}, ...]
+      [%Enterprise{}, ...]
 
   """
-  @spec get_connected_hubs() :: connected_hubs()
+  @spec get_connected_hubs() :: list(Provider.t())
   def get_connected_hubs do
-    for hub <- get_hubs(), connected = get_connected_hub(hub), do: connected
+    for hub <- get_hubs(),
+        capability?(hub, :connect),
+        Provider.connected?(hub),
+        into: [],
+        do: hub
+  end
+
+  @doc """
+  Gets a list of connected hubs with given capabilities.
+
+  ## Example
+
+      iex> get_connected_hubs([:secrets])
+      [%Enterprise{}, ...]
+
+  """
+  @spec get_connected_hubs(Provider.capabilities()) :: list(Provider.t())
+  def get_connected_hubs(capabilities) do
+    for hub <- get_connected_hubs(),
+        capability?(hub, capabilities -- [:connect]),
```

Comment:
Quoting @wojtekmach [review](https://github.com/livebook-dev/livebook/pull/1663#discussion_r1090451684):

> also, why do we need to reject :connect?

Because I'm already validating it with `get_connected_hubs/0`, which only returns all hubs that can connect and they are already connected.

Reply:
OK, got it, thanks. IMHO not worth it, doesn't hurt to do this redundant check, but I don't have a strong opinion. Out of curiosity do we ever call it with :connect?

Reply:
We don't call it with `:connect`, but if someone passes, we would ignore.
Doesn't hut, as you said, but it would sound like "duplicated", but yeah, I don't have a strong opinion too 😬

Reply:
Alternatively you could make it `get_connected_hubs(opts \\ [])`, then pass capabilities as an option and always add `:connect` to those capabilities.

Reply:
This function doesn't exist anymore, so it's okay 😆